pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -101,15 +101,16 @@ def restore(self, checkpoint_path: str, on_gpu: bool):
         # load model state
         model = self.trainer.get_model()
 
-        # load the state_dict on the model automatically
-        model.load_state_dict(checkpoint['state_dict'])
-
         # give the datamodule a chance to load something
         if self.trainer.datamodule is not None:
             self.trainer.datamodule.on_load_checkpoint(checkpoint)
+
         # give model a chance to load something
         model.on_load_checkpoint(checkpoint)
 
+        # load the state_dict on the model automatically
+        model.load_state_dict(checkpoint['state_dict'])
```

Comment:
```suggestion
        model.load_state_dict(checkpoint['state_dict'], strict=checkpoint.get('strict', True))
```

will it be good to have this? one can add `strict=False`(default=True) to `checkpoint` in `on_load_checkpoint`. We do have `strict` parameter in `load_from_checkpoint`.

Reply:
i don't think it's needed. the user can either alter checkpoint in on_load_checkpoint, or they can override load_state_dict in the lightning module. wdyt?

Reply:
altering checkpoint in `on_load_checkpoint` just to avoid `strict=True` maybe difficult. Let's say if I add more layers to my model, then I have to add weights in state_dict too which is stored in `checkpoint`, although I have never seen a case where model is altered while resuming training but this is just an additional feature that might be useful.

Also I don't think user can modify `load_state_dict`.